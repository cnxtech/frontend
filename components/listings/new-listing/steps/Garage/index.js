import React, { Component } from 'react'
import { Formik, Field } from 'formik'

import Button from '@emcasa/ui-dom/components/Button'
import RadioButton from '@emcasa/ui-dom/components/RadioButton'
import Row from '@emcasa/ui-dom/components/Row'
import Col from '@emcasa/ui-dom/components/Col'
import View from '@emcasa/ui-dom/components/View'
import Text from '@emcasa/ui-dom/components/Text'
import NavButtons from 'components/listings/new-listing/shared/NavButtons'

class Garage extends Component {
  constructor(props) {
    super(props)
    this.nextStep = this.nextStep.bind(this)
    this.previousStep = this.previousStep.bind(this)
    this.validateSpots = this.validateSpots.bind(this)
    this.validateSpotType = this.validateSpotType.bind(this)
    this.updateStateFromProps = this.updateStateFromProps.bind(this)
  }

  state = {
    spots: null,
    spotType: null
  }

  componentDidMount() {
    this.updateStateFromProps(this.props)
  }

  componentWillReceiveProps(props) {
    this.updateStateFromProps(props)
  }

  updateStateFromProps(props) {
    const { garage } = props
    if (garage) {
      this.setState({
        spots: garage.spots,
        spotType: garage.spotType
      })
    }
  }

  nextStep() {
    const { navigateTo, updateGarage } = this.props
    updateGarage(this.state)
    navigateTo('differential')
  }

  previousStep() {
    const { navigateTo } = this.props
    navigateTo('bedrooms')
  }

  validateSpots(spots) {
    if (typeof spots !== 'number') {
      return "É necessário informar o número de vagas"
    }
  }

  validateSpotType(spotType) {
    if (this.state.spots > 0 && !spotType) {
      return "É necessário informar o tipo de vaga"
    }
  }

  render() {
    const { garage } = this.props
    let spots, spotType
    if (garage) {
      spots = garage.spots
      spotType = garage.spotType
    }
    return (
      <div ref={this.props.hostRef}>
        <Row justifyContent="center">
          <Col width={[1, 1/2]}>
            <Formik
              initialValues={{
                spots: spots,
                spotType: spotType
              }}
              isInitialValid={() => {
                return !(this.validateSpots(spots) && this.validateSpotType(spotType))
              }}
              render={({isValid, setFieldTouched, setFieldValue, errors}) => (
                <>
                  <View body p={4}>
                    <Text
                      fontSize="large"
                      fontWeight="bold"
                      textAlign="center">
                      Tem vagas de garagem?
                    </Text>
                    <Text color="grey">Seu imóvel tem vagas de garagem?</Text>
                    <Row mb={4}>
                      <Field
                        name="spots"
                        validate={this.validateSpots}
                        render={() =>
                          <Button.Group flexWrap="wrap" initialValue={spots} onChange={(value) => {
                            setFieldValue('spots', value)
                            setFieldTouched('spots')
                            this.setState({spots: value})
                            }}>
                            <Button mr={2} value={0} height="tall">Não tem</Button>
                            <Button mr={2} value={1} height="tall">1</Button>
                            <Button mr={2} value={2} height="tall">2</Button>
                            <Button mr={2} value={3} height="tall">3</Button>
                            <Button mr={2} value={4} height="tall">4</Button>
                          </Button.Group>
                        }/>
                    </Row>
                    {this.state.spots > 0 && <>
                      <Text color="grey">As vagas estão na Escritura do imóvel ou são do Condomínio?</Text>
                      <Row mb={4}>
                        <Col width={1}>
                          <Field
                            name="spotType"
                            validate={this.validateSpotType}
                            render={() => (
                              <RadioButton.Group
                                selectedValue={this.state.spotType}
                                onChange={(value) => {
                                  setFieldValue('spotType', value)
                                  setFieldTouched('spotType')
                                  this.setState({spotType: value})
                                }}>
                                <RadioButton label="Vagas na Escritura" value="deed"/>
                                <View mb={2}></View>
                                <RadioButton label="Vagas no Condomínio" value="condominium" />
                              </RadioButton.Group>
                            )} />
                        </Col>
                      </Row>
                    </>}
                  </View>
                  <View bottom p={4}>
                    <NavButtons
                      previousStep={this.previousStep}
                      onSubmit={this.nextStep}
                      submitEnabled={isValid}
                    />
                  </View>
                </>
              )}
            />
          </Col>
        </Row>
      </div>
    )
  }
}

export default Garage
