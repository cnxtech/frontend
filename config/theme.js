import theme from '@emcasa/ui'
import {fas} from '@fortawesome/pro-solid-svg-icons'
import {fal} from '@fortawesome/pro-light-svg-icons'
import icons from './icons'

export default {
  ...theme,
  icons: {emcasa: icons, default: fas, light: fal}
}
