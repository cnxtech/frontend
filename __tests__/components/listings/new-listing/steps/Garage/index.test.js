import React from 'react'
import Garage from 'components/listings/new-listing/steps/Garage'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import theme from '@emcasa/ui'
import { ThemeProvider } from 'styled-components'

it('renders correctly', () => {
  const tree = renderer.create(
    <ThemeProvider theme={theme}>
      <Garage />
    </ThemeProvider>).toJSON()
  expect(tree).toMatchSnapshot()
})
