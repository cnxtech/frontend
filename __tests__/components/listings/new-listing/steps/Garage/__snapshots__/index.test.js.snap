// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 20px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 20px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 20px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c6 {
  outline: none;
  cursor: pointer;
  padding: 0 20px 0 20px;
  border-radius: 4px;
  border: 1px solid;
  border-color: #e0e6ed;
  background-color: #ffffff;
  width: auto;
  height: 60px;
  margin-right: 10px;
  padding-left: 14px;
  padding-right: 14px;
  color: #3c4858;
  font-family: FaktSoftPro-Normal;
  font-size: 18px;
  -webkit-font-smoothing: antialiased;
  border-color: #e0e6ed;
}

.c6 :hover {
  background-color: #ffffff;
}

.c9 {
  outline: none;
  display: block;
  cursor: pointer;
  padding: 0 20px 0 20px;
  border-radius: 4px;
  border: 1px solid;
  border-color: #e0e6ed;
  background-color: #ffffff;
  width: 100%;
  height: 60px;
  color: #3c4858;
  font-family: FaktSoftPro-Normal;
  font-size: 18px;
  -webkit-font-smoothing: antialiased;
  border-color: #e0e6ed;
}

.c9 :hover {
  background-color: #ffffff;
}

.c10 {
  outline: none;
  display: block;
  cursor: default;
  padding: 0 20px 0 20px;
  border-radius: 4px;
  border: 1px solid;
  border-color: #e0e6ed;
  background-color: #ffffff;
  width: 100%;
  height: 60px;
  color: #8492a6;
  font-family: FaktSoftPro-Normal;
  font-size: 18px;
  -webkit-font-smoothing: antialiased;
  border-color: #e0e6ed;
}

.c10 :hover {
  background-color: #ffffff;
}

.c1 {
  width: 100%;
}

.c8 {
  width: 41.66666666666667%;
}

.c2 {
  -webkit-font-smoothing: antialiased;
  font-family: FaktSoftPro-Normal;
  color: #3c4858;
  font-size: 24px;
  font-weight: bold;
  text-align: center;
}

.c3 {
  -webkit-font-smoothing: antialiased;
  font-family: FaktSoftPro-Normal;
  color: #979797;
  font-size: 18px;
}

@media screen and (min-width:640px) {
  .c1 {
    width: 50%;
  }
}

<div>
  <div
    className="c0"
  >
    <div
      className="c1"
      width={
        Array [
          1,
          0.5,
        ]
      }
    >
      <p
        className="c2"
        color="dark"
        fontSize="large"
        fontWeight="bold"
      >
        Tem vagas de garagem?
      </p>
      <p
        className="c3"
        color="grey"
        fontSize="default"
      >
        Seu imóvel tem vagas de garagem?
      </p>
      <div
        className="c4"
      >
        <div
          className="c5"
          onChange={[Function]}
        >
          <button
            className="c6"
            fontSize="default"
            height="tall"
            onClick={[Function]}
            selected={false}
            value={0}
          >
            Não tem
          </button>
          <button
            className="c6"
            fontSize="default"
            height="tall"
            onClick={[Function]}
            selected={false}
            value={1}
          >
            1
          </button>
          <button
            className="c6"
            fontSize="default"
            height="tall"
            onClick={[Function]}
            selected={false}
            value={2}
          >
            2
          </button>
          <button
            className="c6"
            fontSize="default"
            height="tall"
            onClick={[Function]}
            selected={false}
            value={3}
          >
            3
          </button>
          <button
            className="c6"
            fontSize="default"
            height="tall"
            onClick={[Function]}
            selected={false}
            value={4}
          >
            4
          </button>
        </div>
      </div>
      <div
        className="c7"
      >
        <div
          className="c8"
          width={0.4166666666666667}
        >
          <button
            className="c9"
            fontSize="default"
            height="tall"
            onClick={[Function]}
          >
            Voltar
          </button>
        </div>
        <div
          className="c8"
          width={0.4166666666666667}
        >
          <button
            className="c10"
            disabled={true}
            fontSize="default"
            height="tall"
            onClick={[Function]}
          >
            Avançar
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
